src/capture/activated/mod.rs

```diff
@@ -199,6 +201,28 @@ impl<T: Activated + ?Sized> Capture<T> {
         PacketIter::new(self, codec)
     }
 
+    pub fn for_each<F>(&mut self, handler: F)
+    where
+        F: FnMut(Packet),
+    {
+        let mut handler = Handler {
+            func: AssertUnwindSafe(handler),
+            panic_payload: None,
+            handle: self.handle,
+        };
+        unsafe {
+            raw::pcap_loop(
+                self.handle.as_ptr(),
+                -1,
```

Comment:
To prevent API changes going forcing a major version change, can I ask you to expose `cnt` as well? My preference would be to have an enum consisting of two possible options: `Infinite` and `Finite(usize)`. Then inside `for_each` the value can be suitably cast.

Reply:
This will also mean you will need to check the return value of `pcap_loop` (https://www.tcpdump.org/manpages/pcap_loop.3pcap.html). You don't need to handle `PCAP_ERROR_BREAK` as you're not adding `pcap_breakloop` support. `PCAP_ERROR_NOT_ACTIVATED` can be assumed impossible due to the `Capture` type. However, `PCAP_ERROR` should be handled. You can use the `self.check_err` or `self.get_err` as appropriate.

Reply:
I originally proposed a new enum`Infinite` and `Finite(usize)` since at first `None` didn't make sense to me for infinite packets. However, on reflection the variable name is `count` so `None` means "no count". An `Option<usize>` might be better than a new enum. Plus it comes with all the various methods implemented for it.